test/unittest/DatabaseQueue/CMakeLists.txt

```diff
@@ -27,18 +27,25 @@ if(GTEST_FOUND AND GMOCK_FOUND)
     ${PROJECT_SOURCE_DIR}/src/cpp/database/samples.cpp
     ${PROJECT_SOURCE_DIR}/src/cpp/database/database_queue.cpp)
 
+    if(MSVC)
+        target_compile_definitions(database_queue_tests
+            PRIVATE _CRT_DECLARE_NONSTDC_NAMES=0 FASTDDS_STATISTICS_BACKEND_SOURCE)
+    endif(MSVC)
+
     target_include_directories(database_queue_tests PRIVATE
         ${GTEST_INCLUDE_DIRS} ${GMOCK_INCLUDE_DIRS}
         ${PROJECT_SOURCE_DIR}/test/mock/database/database
         ${PROJECT_SOURCE_DIR}/test/mock/StatisticsBackend
         ${PROJECT_SOURCE_DIR}/include
-        ${PROJECT_SOURCE_DIR}/include/${PROJECT_NAME}
+        ${PROJECT_SOURCE_DIR}/include/${PROJECT_NAME_UNDERSCORED}
         ${PROJECT_BINARY_DIR}/include
-        ${PROJECT_BINARY_DIR}/include/${PROJECT_NAME}
+        ${PROJECT_BINARY_DIR}/include/${PROJECT_NAME_UNDERSCORED}
         ${PROJECT_SOURCE_DIR}/src/cpp)
 
     target_link_libraries(database_queue_tests PUBLIC
-        ${GTEST_LIBRARIES} ${GMOCK_LIBRARIES} fastrtps)
+        ${GTEST_LIBRARIES} ${GMOCK_LIBRARIES} fastrtps fastcdr)
```

Comment:
I don't think we need fastcdr in this test

Reply:
Any test building `types.cxx` will require fastcdr